src/realm/sort_descriptor.cpp

```diff
@@ -591,3 +606,47 @@ void DescriptorOrdering::get_versions(const Group* group, TableVersions& version
         versions.emplace_back(table_key, group->get_table(table_key)->get_content_version());
     }
 }
+
+std::string SemanticSearchDescriptor::get_description(ConstTableRef) const
+{
+    return "KNN()";
+}
+
+void SemanticSearchDescriptor::execute(const Table& table, KeyValues& key_values) const
+{
+    if (m_k >= key_values.size()) return; // all entries already match as closest
+    
+    std::priority_queue<std::pair<float, ObjKey>> topResults;
+    
+    // Collect the k closest matches by distance
+    size_t n = std::min(m_k, key_values.size());
```

Comment:
Why min? Why there are two loop? Shouldn't this just go through all key_values once and find m_k closest for m_query_data? Either i don't quite understand something, or this is a bit off.

Reply:
ah, next loop starts from m_k and priority_queue is max queue by default. Keep forgetting this, all good here.